apps/web/src/pages/Users/UpdateUserPage/components/TeamRoleTable.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useMemo } from "react";
+import { useIntl } from "react-intl";
+
+import { notEmpty } from "@gc-digital-talent/helpers";
+import { Heading, Link, Pill } from "@gc-digital-talent/ui";
+import { getLocalizedName } from "@gc-digital-talent/i18n";
+
+import Table, { ColumnsOf, Cell } from "~/components/Table/ClientManagedTable";
+import {
+  Role,
+  RoleAssignment,
+  UpdateUserAsAdminInput,
+  User,
+} from "~/api/generated";
+import useRoutes from "~/hooks/useRoutes";
+
+import AddTeamRoleDialog from "./AddTeamRoleDialog";
+import RemoveTeamRoleDialog from "./RemoveTeamRoleDialog";
+import { UpdateUserFunc } from "../types";
+
+const roleCell = (displayName: string) => {
+  return (
+    <Pill color="neutral" mode="solid">
+      {displayName}
+    </Pill>
+  );
+};
+
+const actionCell = (
+  roleAssignment: RoleAssignment,
+  user: User,
+  onUpdateUser: UpdateUserFunc,
+) => {
+  return roleAssignment.role && user && roleAssignment.team ? (
+    <RemoveTeamRoleDialog
+      role={roleAssignment.role}
+      user={user}
+      team={roleAssignment.team}
+      onUpdateUser={onUpdateUser}
+    />
+  ) : null;
+};
+
+const teamCell = (displayName: string, href: string) => {
```

Comment:
gave multiple team based roles a try

![image](https://user-images.githubusercontent.com/40485260/227647466-e647f3e0-54aa-42aa-8fa4-10889435cdac.png)

and it seems to work fine! nice!

https://recordit.co/A3n0sNEbfO

Reply:
things look good to me, I want to let Eric/Pat have the final say since they worked on this a lot and may have code improvement suggestions